docs/manual/creating-a-checker.tex

```diff
@@ -1182,29 +1182,77 @@
-The methods operate on \refclass{framework/type}{AnnotatedTypeMirror},
-which is the Checker Framework's representation of an annotated type.
-The methods can make arbitrary changes to the annotations on a type.
+can write your own type introduction rules.  There are three ways to do so.
+
+\begin{enumerate}
+\item
+  Define a subclass of
+  \refclass{org/checkerframework/framework/type/treeannotator}{TreeAnnotator},
+  typically as a private inner class of your \<AnnotatedTypeFactory>.
+  There is a method of \<TreeAnnotator> for every AST node, and the visitor
+  has access to both the tree (the AST node) and its type.  In your
+  subclass of \<AnnotatedTypeFactory>, override \<createTreeAnnotator> to
+  return a \<ListTreeAnnotator> containing that annotator, as in
 
-%TODO: document tree and type annotators here, which should be used instead of override addComputedTypeAnnotations if possible.
+\begin{Verbatim}
+  @Override
+  protected TreeAnnotator createTreeAnnotator() {
+      return new ListTreeAnnotator(super.createTreeAnnotator(), new MyTreeAnnotator(this));
+  }
+\end{Verbatim}
+
+  \noindent
+  (or put your TreeAnnotator first; note that \<PropagationTreeAnnotator>,
+  which is run by default, adds annotations to unannotated trees, but has
+  no effect on trees that have an annotation).
+
+\item
+  Define a subclass of a
+  \refclass{org/checkerframework/framework/type/treeannotator}{TypeAnnotator},
+  typically as a private inner class of your \<AnnotatedTypeFactory>.
+  There is a method of \<TypeAnnotator> for every kind of type, and the
+  visitor has access to only the type.  In your subclass of
+  \<AnnotatedTypeFactory>, override \<createTypeAnnotator> to return a
+  \<ListTypeAnnotator> containing that annotator, as in
+
+\begin{Verbatim}
+  @Override
+  protected TypeAnnotator createTypeAnnotator() {
+      return new ListTypeAnnotator(new MyTypeAnnotator(this), super.createTypeAnnotator());
+  }
+\end{Verbatim}
+
+  \noindent
+  (or put your TypeAnnotator last).
+
+\item
+  Create a subclass of \refclass{framework/type}{AnnotatedTypeFactory} and
+  override its two \<addComputedTypeAnnotations> methods.  This is a last
+  resort, if your logic cannot be implemented using a TreeAnnotator or a
+  TypeAnnotator.  The implementation of \<addComputedTypeAnnotations> in
+  \<GenericAnnotatedTypeFactory> calls the tree annotator and the type
+  annotator (in that order), but by overriding the method you can cause
+  your logic to be run even earlier or even later.
+
+  \<AnnotatedTypeFactory>, when given a program
+  expression, returns the expression's type.  This should include not only
+  the qualifiers that the programmer explicitly wrote in the source code, but
+  also default annotations and type
+  refinement (see Section~\ref{effective-qualifier} for explanations of these
+  concepts).
+
+  To add type introduction rules, you should override
```

Comment:
I think you should have deleted the rest of this section?

Reply:
That text is unchanged from the previous manual (just indented).  I think it adds information and should be retained.  But maybe I'm missing a reason it should be deleted.  Could you clarify?

Reply:
I noticed it is unchanged, that's why I thought you meant to delete it.  The text reads as if it is the only way to add type introduction rules when it is just one of three ways to do so.  It's helpful to explicitly state the methods to override. 

I think you can just delete the sentence below or move it earlier in the section.  The other two items in the lists assume the read knows what an `AnnotatedTypeMirror` is. 
>  The methods operate on \refclass{framework/type}{AnnotatedTypeMirror},
  which is the Checker Framework's representation of an annotated type.